pkg/upgrade/upgrade_test.go

```
@@ -0,0 +1,186 @@
+	require.NoError(t, err, "Failed to start log streaming")
+	defer cancel()
+
+	timeout := time.After(5 * time.Second)
+
+	for {
+		select {
+		case line, ok := <-logChan:
+			if !ok {
+				t.Fatalf("Log stream closed before finding target log")
+			}
+			t.Logf(line)
```

golangci-lint (1.62.2)

printf: non-constant format string in call to (*testing.common).Logf

(govet)